src/react/components.js

```diff
@@ -284,24 +284,29 @@ export function evaluateClassConstructor(
   let instanceProperties = new Set();
   let instanceSymbols = new Set();
 
-  realm.evaluatePure(() =>
-    realm.evaluateForEffects(
-      () => {
-        let instance = Construct(realm, constructorFunc, [props, context]);
-        invariant(instance instanceof ObjectValue);
-        for (let [propertyName] of instance.properties) {
-          if (!whitelistedProperties.has(propertyName)) {
-            instanceProperties.add(propertyName);
+  realm.evaluatePure(
+    () =>
+      realm.evaluateForEffects(
+        () => {
+          let instance = Construct(realm, constructorFunc, [props, context]);
+          invariant(instance instanceof ObjectValue);
+          for (let [propertyName] of instance.properties) {
+            if (!whitelistedProperties.has(propertyName)) {
+              instanceProperties.add(propertyName);
+            }
           }
-        }
-        for (let [symbol] of instance.symbols) {
-          instanceSymbols.add(symbol);
-        }
-        return instance;
-      },
-      /*state*/ null,
-      `react component constructor: ${constructorFunc.getName()}`
-    )
+          for (let [symbol] of instance.symbols) {
+            instanceSymbols.add(symbol);
+          }
+          return instance;
+        },
+        /*state*/ null,
+        `react component constructor: ${constructorFunc.getName()}`
+      ),
+    // We don't want to track mutations on "this" but of the outside
+    // context of the constructor
+    true,
```

Comment:
I changed this to `false` and none of the tests failed. I know you have a long comment explaining what this does, but it didn't really help me. Can you show a practical example for why it matters?

Reply:
I'll come up with a test does fail when it's false.

Reply:
We can leave it off, you're right that this doesn't help here. This functionality is mainly for the work in the other PR that adds pure scope checking around all function calls when in pure mode.